### PR TITLE
CI: Test on MSRV, stable, beta and nightly Rust versions

### PR DESCRIPTION
**Pull request by weiji14**
Add test matrix to test on minimum supported Rust version (1.70), stable, beta and nightly.

- [x] I agree to follow the project's [code of conduct](https://github.com/georust/.github/blob/main/CODE_OF_CONDUCT.md).
- [ ] I added an entry to the project's change log file if knowledge of this change could be valuable to users.
  - Usually called `CHANGES.md` or `CHANGELOG.md`
  - Prefix changelog entries for breaking changes with "BREAKING: "
---

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -16,13 +16,23 @@ permissions:
...
     steps:
-    - name: Checkout repository
-      uses: actions/checkout@v4
+      - name: Checkout repository
+        uses: actions/checkout@v4
 
-    - name: Build
-      run: cargo build --verbose
+      - name: Update Rust toolchain
+        run: rustup update ${{ matrix.toolchain }} && rustup default ${{ matrix.toolchain }}
```
Comment (kylebarron) [inline .github/workflows/test.yml:32]: use https://github.com/dtolnay/rust-toolchain?

**Review by kylebarron — APPROVED** [review]

File: .github/workflows/test.yml (hunk 1 above)
Comment (weiji14) [reply .github/workflows/test.yml:32]: I'm slightly paranoid with relying on 3rd party GitHub Actions after the recent `tj-actions/changed-files` compromise, probably ok to use the built-in `rustup` here for now.
